pluggy/manager.py

```diff
@@ -289,6 +289,11 @@ def get_hookcallers(self, plugin):
         """ get all hook callers for the specified plugin. """
         return self._plugin2hookcallers.get(plugin)
 
+    def get_hookimpl(self, name):
+        """ Return a list of all hook implementations (HookImpl) for a given hook. """
+        hc = getattr(self.hook, name)
+        return hc._wrappers + hc._nonwrappers
```

Comment:
the part dealing with private attributes should be delegated to the hookcaller

Reply:
seems like i missed the intorduction of the patter, lets do this as a followup issue so this one canb e self-contained an in line with the current coding standard, we can resolve the pattern as a separate pr after its merit is taken into account

Reply:
Ok!

Another thing - if hookcaller can give you the hookimpl, then maybe it is more general if we can just get the caller from pluginmanager and then  call `get_hookimpl()` on the returned caller.

For getting the caller, we have two options:
- the plugin manager just have a `hook_caller(name)`. Reads nice and simple.
- or actually just use `subset_hook_caller` with `remove_plugins` as `None` or `[]`. It actually just return the original caller. So need to introduce another function, but does not read as nice (running this *subset* method to actually get the whole thing, not the subset)

I think I prefer the first option.

Reply:
for now i wont explore the domain in detail as i lack personal time